segmentstore/server/src/test/java/io/pravega/segmentstore/server/store/StreamSegmentContainerRegistryTests.java

```diff
@@ -207,28 +241,31 @@ public int getId() {
 
         @Override
         public void close() {
-            if (!this.closed) {
+            if (!this.closed.getAndSet(true)) {
                 FutureHelpers.await(ServiceHelpers.stopAsync(this, ForkJoinPool.commonPool()));
-                this.closed = true;
             }
         }
 
         @Override
         protected void doStart() {
-            if (this.startException != null) {
-                notifyFailed(this.startException);
-            } else {
-                notifyStarted();
-            }
+            ForkJoinPool.commonPool().execute(() -> {
```

Comment:
Actually I just realized we are inheriting from `ThreadPooledTestSuite`. I changed this to use the provided Executor, and not the `ForkJoinPool`.